smslibrary/src/main/java/com/eis/smslibrary/SMSPeer.java

```diff
@@ -4,56 +4,112 @@
+     * @return The PhoneNumber object containing the phone number parsed.
+     * @throws IllegalArgumentException If the given destination is not parsable.
+     * @author Matteo Carnelos
      */
-    public static TelephoneNumberState checkPhoneNumber(String telephoneNumber) {
-        //Check if the number is shorter than the MAX.
-        if (telephoneNumber.length() > MAX_TELEPHONE_NUMBER_LENGTH + 1) {
-            return TelephoneNumberState.TELEPHONE_NUMBER_TOO_LONG;
-        }
-        //Check if the number is longer than the MIN.
-        if (telephoneNumber.length() < MIN_TELEPHONE_NUMBER_LENGTH + 1) {
-            return TelephoneNumberState.TELEPHONE_NUMBER_TOO_SHORT;
-        }
-        //Check if it's actually a number with a plus and doesn't contain anything else
-        if (!telephoneNumber.matches("\\+?\\d{"+MIN_TELEPHONE_NUMBER_LENGTH+","+MAX_TELEPHONE_NUMBER_LENGTH+"}")) {
-            return TelephoneNumberState.TELEPHONE_NUMBER_NOT_A_NUMBER;
+    private Phonenumber.PhoneNumber parseNumber(String number) throws IllegalArgumentException {
+        Phonenumber.PhoneNumber phoneNumber;
+        try { phoneNumber = PhoneNumberUtil.getInstance().parse(number, defaultRegion); }
+        catch(NumberParseException e) {
+            throw new IllegalArgumentException("Not parsable telephone number: \"" + number + "\".");
         }
-        //Check if there is a country code.
-        if (telephoneNumber.charAt(0) != '+') {
-            return TelephoneNumberState.TELEPHONE_NUMBER_NO_COUNTRY_CODE;
+        return phoneNumber;
+    }
+
+    /**
+     * Format the given phone number into the E164 format. The E164 format is basically the string
+     * address without dashes, spaces and brackets. Example: "+393401234567".
+     *
+     * @param phoneNumber The {@link Phonenumber.PhoneNumber} object.
+     * @return A string representing the formatted number.
+     * @author Matteo Carnelos
+     */
+    private String formatE164(Phonenumber.PhoneNumber phoneNumber) {
+        return PhoneNumberUtil.getInstance().format(phoneNumber, PhoneNumberUtil.PhoneNumberFormat.E164);
+    }
+
+    /**
+     * Adjust the address string in case it is the phone number of an emulator.
+     * The emulator phone numbers could be either "+1555521555*", "555521555*" or "555*". All
+     * these numbers are invalid in the real world, so they need to avoid validation checks.
+     *
+     * @param address The address to eventually adjust.
+     * @return True if the number given is an emulator number, false otherwise.
+     * @author Matteo Carnelos
+     */
+    private boolean adjustForEmulator(String address) {
+        // This regex checks if the given number is one of the following types:
+        // +1#555521555*     [* = any decimal number]
+        // 555521555*        [# = either a space or '-']
+        // +1#555*
+        // 555*
+        if(address.matches("^(\\+1)?[ -]?(555521)?555\\d$")) {
```

Comment:
Regex could be saved as static final field too

Reply:
Yes, you're right